### PR TITLE
adding initial map type support, and additional housekeeping

### PR DESCRIPTION
**Pull request by burhan94**
This commit adds initial map type support to the sdk. 

Note: none of the connectors have added this support yet. Each connector will need a separate implementation.

### CODE REVIEW

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/serde/v2/ArrowTypeSerDe.java [hunk 1]
```diff
@@ -684,4 +686,40 @@ protected ArrowType doTypedDeserialize(JsonParser jparser, DeserializationContex
...
+            private Deserializer()
+            {
+                super(ArrowType.class, ArrowType.Map.class);
+            }
+
+            @Override
+            protected ArrowType doTypedDeserialize(JsonParser jparser, DeserializationContext ctxt)
+                    throws IOException
+            {
+                return new ArrowType.Map(false);
```
Comment (jobrandw) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/serde/v2/ArrowTypeSerDe.java:721]: What does passing `false` to the constructor do?  Might be worth adding this as a constant with more context.

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java [hunk 2]
```diff
@@ -1031,6 +1053,271 @@ else if (value instanceof Boolean && (boolean) value) {
...
+            if (!KEY.equals(keyField.getName()) || !VALUE.equals(valueField.getName())) {
+                throw new IllegalStateException("Invalid Arrow Map schema: " + field);
+            }
+        }
+
+        for (Field nextChild : keyValueChildren) {
+            //For each child field that comprises the Map, attempt to extract and write the corresponding value
+            //using the FieldResolver.
+            Object childValue = resolver.getFieldValue(nextChild, value);
+            switch (Types.getMinorTypeForArrowType(nextChild.getType())) {
```
Comment (jobrandw) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java:1113]: Do we need to support nested maps?

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java [hunk 3]
```diff
@@ -1031,6 +1053,271 @@ else if (value instanceof Boolean && (boolean) value) {
...
+    /**
+     * Used to write a value into a specific child field within a Map. Multiple calls to this method per-cell are
+     * expected in order to write to all N fields of a Map.
+     *
+     * @param writer The FieldWriter (already positioned at the row and list entry number) that we want to write into.
+     * @param field The child field we are attempting to write into.
+     * @param allocator The BlockAllocator that can be used for allocating Arrow Buffers for fields which require conversion
+     * to Arrow Buff before being written.
+     * @param value The value to write.
+     * @note This method and its List complement violate the DRY mantra because ListWriter and StructWriter don't share
```
Comment (jobrandw) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java:1147]: do you mean ListWriter and _UnionMapWriter_?

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/serde/v2/ArrowTypeSerDe.java (hunk 1 above)
Comment (burhan94) [reply athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/serde/v2/ArrowTypeSerDe.java:721]: false just means that the entries are not ordered. I can add a comment.

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java (hunk 2 above)
Comment (burhan94) [reply athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java:1113]: not at the moment, this is for basic maps. We can enhance later on.

Comment (burhan94) [reply athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java:1113]: We should. Maps with struct or list values are a valid case.

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java (hunk 3 above)
Comment (burhan94) [reply athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java:1147]: oops, typo.

**Review by jobrandw — APPROVED** [review]
lgtm

**Pull request merged**